### PR TITLE
Add integration tests #2388

### PR DESCRIPTION
**Pull request by zbendhiba**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java [hunk 1]
```diff
@@ -184,35 +194,362 @@ private Action doAction(int id, DigitalOceanOperations operation) {
...
+        headers.put(DigitalOceanHeaders.OPERATION, DigitalOceanOperations.get);
+        headers.put(DigitalOceanHeaders.ID, volumeId);
+        headers.put(DigitalOceanHeaders.REGION, REGION_FRANKFURT);
+        return producerTemplate.requestBodyAndHeaders("direct:blockStorages", null, headers, Volume.class);
+    }
+
+    @POST
+    @Path("blockStorages/attach/{name}")
+    public Action attachVolumeToDroplet(@PathParam("name") String volumeName, String dropletId) {
+        LOG.infof("attache volume %s to droplet %s in region %s", volumeName, dropletId, REGION_FRANKFURT);
```
Comment (aldettinger) [inline integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java:426]: Minor typo 'attache'

File: integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java [hunk 2]
```diff
@@ -184,35 +194,362 @@ private Action doAction(int id, DigitalOceanOperations operation) {
...
+    public List<Volume> getAvailableVolumes() {
+        LOG.infof("get all avalaible volumes for region %s", REGION_FRANKFURT);
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(DigitalOceanHeaders.OPERATION, DigitalOceanOperations.list);
+        headers.put(DigitalOceanHeaders.REGION, REGION_FRANKFURT);
+        return producerTemplate.requestBodyAndHeaders("direct:blockStorages", null, headers, List.class);
+    }
+
+    @DELETE
+    @Path("blockStorages")
```
Comment (aldettinger) [inline integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java:458]: "blockStorages" path looks duplicated

File: integration-tests/digitalocean/src/test/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanDropletTest.java [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+    }
+
+    /**
+     * Assert all the actions
+     *
+     */
+    private void assertActions(List<Map<String, Object>> actions, List<String> types) {
+        // verify there are actions
+        assertNotNull(actions);
+        // verify there are at least the 7 created actions in the test
```
Comment (aldettinger) [inline integration-tests/digitalocean/src/test/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanDropletTest.java:228]: So, I guess it's not always 7 in the context of this function.

**Review by aldettinger — APPROVED** [review]
Minor issues, it looks good to me. The comments really help to understand the test logic.
The build failure is unrelated and I was able to build locally.

**Review by jamesnetherton — APPROVED** [review]

File: integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java (hunk 1 above)
Comment (zbendhiba) [reply integration-tests/digitalocean/src/main/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanResource.java]: @aldettinger I've just pushed the fix typo. the only thing that I've changed

File: integration-tests/digitalocean/src/test/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanDropletTest.java (hunk 3 above)
Comment (zbendhiba) [reply integration-tests/digitalocean/src/test/java/org/apache/camel/quarkus/component/digitalocean/it/DigitaloceanDropletTest.java:228]: Yes there can be some other actions done by Digitalocean for instance (internal process); So I was just testing the actions that we know we've done in this test suit

**Comment by zbendhiba** [comment]
@aldettinger so can I merge this one ?

**Comment by aldettinger** [comment]
Sure, I see no blockers so this PR get my approval :+1:

**Pull request merged**